src/logic/calculate.js

```diff
@@ -0,0 +1,13 @@
+export default function calculate(calculatorData, buttonName) {
+  let { total, next } = calculatorData;
+  const { operation } = calculatorData;
```

Comment:
### [Optional]
I think we could have this destructuring in one line;
`  let { total, next, operation } = calculatorData;`